lib/dpdk/iceBoardVDIF.hpp

```diff
@@ -301,19 +307,11 @@ void iceBoardVDIF::copy_packet_vdif(struct rte_mbuf* mbuf) {
                     from_idx -= mbuf_len; // Subtract the last mbuf_len from the current idx.
                     mbuf_len = mbuf->data_len;
                 }
-
-                int output_idx =
-                    vdif_frame_location * frame_size +           // Frame location in output buffer.
-                    vdif_packet_len * num_elements * time_step + // Time step in output frame.
-                    vdif_packet_len * elem + // VDIF pack for the correct element (ThreadID).
-                    vdif_header_len +        // Offset for the vdif header.
-                    tel.to_freq_id(encoded_id,
-                                   freq); // Location in the VDIF packet is just frequency.
-
                 // After all that indexing copy one byte :)
                 out_buf_frame[output_idx] = *(rte_pktmbuf_mtod(mbuf, char*) + from_idx);
```

Comment:
I think this could be made faster still with pointer arithmetic, but perhaps the compiler optimizes it already. With it, possibly more than 2 inputs could be saved. Anyway, that definitely seems for later. Next step is probably to ensure we can pick any two arbitrary input channels.

Reply:
The compiler should optimize out `rte_pktmbuf_mtod`, since I believe it's a header level inline function.   If we are getting the performance we need, not obvious it needs to be changed, but might be worth doing if we need to extract more than two inputs, or different patterns of inputs.

Reply:
The thing we have to be careful about is `mbuf` is linked list, so as we make changes, we have to make sure we are still iterating to the next part of that linked list correctly.